### PR TITLE
Use serviceManager to manage timeline button states.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Buttons will be enabled/disabled and selected/not selected based on information provided by ServiceExtensionManager.

### CODE REVIEW

File: lib/timeline/timeline.dart [hunk 1]
```diff
@@ -45,6 +46,17 @@ class TimelineScreen extends Screen {
 
   PButton pauseButton;
   PButton resumeButton;
+  PButton trackWidgetBuildsButton;
+  PButton perfOverlayButton;
+  PButton repaintRainbowButton;
+  PButton debugPaintButton;
+
+  final String trackWidgetBuildsServiceExtension =
```
Comment (jacob314) [inline lib/timeline/timeline.dart]: make these static

File: lib/timeline/timeline.dart [hunk 2]
```diff
@@ -61,13 +73,25 @@ class TimelineScreen extends Screen {
...
-    final PButton trackWidgetBuildsButton = new PButton('Track widget builds')
-      ..small();
-    final PButton perfOverlayButton = new PButton('Performance overlay')
-      ..small();
-    final PButton repaintRainbowButton = new PButton('Repaint rainbow')
-      ..small();
-    final PButton debugDrawButton = new PButton('Debug draw')..small();
+    // TODO(kenzie): perhaps add same icons we use in IntelliJ to these buttons.
+    // This would help to build icon familiarity.
+    PButton createButton(String text, String extensionName) {
```
Comment (jacob314) [inline lib/timeline/timeline.dart]: nit:
createExtensionButton and move this someowhere outside this class as it looks generally useful.

File: lib/timeline/timeline.dart (hunk 1 above)
Comment (kenzieschmoll) [reply lib/timeline/timeline.dart]: Added a service_extensions.dart file to hold const ext names.

File: lib/timeline/timeline.dart (hunk 2 above)
Comment (kenzieschmoll) [reply lib/timeline/timeline.dart]: Done.

File: lib/service_extensions.dart [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const String debugPaintExtName = 'ext.flutter.debugPaint';
```
Comment (jacob314) [inline lib/service_extensions.dart]: Add a library name.

Note that some of our files are missing copyright notices. at some point soon we need to go through and add them everywhere. No rush.

File: lib/service_extensions.dart [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+const String debugPaintBaselinesExtName =
+    'ext.flutter.debugPaintBaselinesEnabled';
+const String repaintRainbowExtName = 'ext.flutter.repaintRainbow';
```
Comment (jacob314) [inline lib/service_extensions.dart]: run dartfmt. I see a missing trailing newline.

File: lib/service_extensions.dart (hunk 3 above)
Comment (kenzieschmoll) [reply lib/service_extensions.dart]: Done.

File: lib/service_extensions.dart [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+const String performanceOverlayExtName = 'ext.flutter.showPerformanceOverlay';
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply lib/service_extensions.dart]: done.

**Comment by jacob314** [comment]
lgtm!

**Pull request merged**